Library/Homebrew/cmd/shellenv.sh

```diff
@@ -77,6 +89,10 @@ homebrew-shellenv() {
       fi
       echo "[ -z \"\${MANPATH-}\" ] || export MANPATH=\":\${MANPATH#:}\";"
       echo "export INFOPATH=\"${HOMEBREW_PREFIX}/share/info:\${INFOPATH:-}\";"
+      if [[ -n "${HOMEBREW_LINUX}" ]]
+      then
+        echo "[ -z \"\${XDG_DATA_DIRS-}\" ] || export XDG_DATA_DIRS=\"${HOMEBREW_PREFIX}/share:\${XDG_DATA_DIRS}\";"
```

Comment:
```suggestion
        echo "[ -z \"\${XDG_DATA_DIRS-}\" ] || export XDG_DATA_DIRS=\"${HOMEBREW_PREFIX}/share:\${XDG_DATA_DIRS-}\";"
```
Just in case some shell behaves weirdly with `set -u`.